tests/ui/generic-associated-types/parse/trait-path-expected-token.stderr

```diff
@@ -5,6 +5,22 @@ LL | fn f1<'a>(arg : Box<dyn X<Y = B = &'a ()>>) {}
    |                               - ^ expected one of 7 possible tokens
    |                               |
    |                               maybe try to close unmatched angle bracket
+   |
+help: expressions must be enclosed in braces to be used as const generic arguments
```

Comment:
Hmmm... this seems pretty strange

Reply:
Well, I'm pretty sure `dyn` is parsed a normal identifier here and not as the edition-2018-specific keyword `dyn` since the test doesn't specify any edition defaulting to 2015. Hence we suggest `{dyn}` and `dyn>` (where `dyn` is interpreted as the name of a type).

Reply:
Not saying that this isn't something I should fix. I will definitely try to improve this output if possible.

Reply:
Is the suggestion better on edition 2021?

Reply:
Yeah, it is:

```
error: expected one of `!`, `(`, `+`, `,`, `::`, `<`, or `>`, found `=`
 --> tests/ui/generic-associated-types/parse/trait-path-expected-token.rs:7:33
  |
7 | fn f1<'a>(arg : Box<dyn X<Y = B = &'a ()>>) {}
  |                               - ^ expected one of 7 possible tokens
  |                               |
  |                               maybe try to close unmatched angle bracket

error: aborting due to previous error
```

Reply:
It may be worthwhile teaching this recovery to ignore `dyn` identifiers (or perhaps all keyword identifiers), even if valid, just for better aesthetics.

Reply:
Will do!

Reply:
Okay, I've gone for a much simpler solution: I only emit the suggestion if `,` or `>` follows the unbraced expression to avoid suggesting garbage if the input is garbage. I've added a small comment for that, too. Now there are no diagnostic regressions at all.